mempool/mempool.go

```diff
@@ -1354,12 +1353,13 @@ func (mp *TxPool) processOrphans(acceptedTx *dcrutil.Tx) []*dcrutil.Tx {
 	var acceptedTxns []*dcrutil.Tx
 
 	// Start with processing at least the passed transaction.
-	processList := list.New()
-	processList.PushBack(acceptedTx)
-	for processList.Len() > 0 {
+	var processList []*dcrutil.Tx
+	processList = append(processList, acceptedTx)
+
+	for len(processList) > 0 {
 		// Pop the transaction to process from the front of the list.
-		firstElement := processList.Remove(processList.Front())
-		processItem := firstElement.(*dcrutil.Tx)
+		processItem := processList[0]
```

Comment:
Since this is house a pointer, you'll need to nil the entry afterwards to be sure the backing array doesn't hold into a reference:

```Go
processItem := processList[0]
processList[0] = nil
processList = processList[1:]
````

Reply:
done